x-pack/platform/packages/shared/kbn-content-packs-schema/src/helpers/index_pattern.ts

```diff
@@ -76,15 +76,14 @@ export function replaceIndexPatterns(
     esqlQuery(query: string) {
       return replaceESQLQueryIndexPattern(query, patternReplacements);
     },
-    indexPattern<T extends { name?: string; title?: string }>(pattern: T) {
+    indexPattern<T extends { title?: string }>(pattern: T) {
       const updatedPattern = pattern.title
         ?.split(',')
         .map((index) => patternReplacements[index] ?? index)
         .join(',');
 
       return {
         ...pattern,
-        name: updatedPattern,
```

Comment:
What is this change about? Was the old version wrong (since it overwrote a potential user-supplied name for the data view)?

Reply:
we were previously updating the name of the saved object with the new index pattern. the rationale was that data view's name can be autofilled with the actual index pattern when created in kibana, for example a `logs*` data view will also be named `logs*` by default. the replacement logic was also applied to the saved object's name, but it was wrong because it replaced the name with the title's content.

I've removed the name replacement because we don't do it anywhere else for now, but could also fix it to replace the proper name property. wdyt ?

Reply:
I see, good question... Happy to keep the current state and see what option works better in practice.

Reply:
i could see the auto naming of things being useful on object creation from the stream UI, but i agree that we shouldn't modify saved object names after creation. i think it could lead to confusion for users.